message.d.ts

```diff
@@ -0,0 +1,588 @@
+     * someFunction(()=>instanceOfKnownProto.serializeBinary());
+     * or, if you don't care about code size, just suppress the
+     * WARNING - Property serializeBinary never defined on Message
+     * and call it the intuitive way.
+     */
+    constructor();
+    private noStructuralTyping____protobuf_javascript_message_Message;
+    static GENERATE_TO_OBJECT: boolean;
+    static GENERATE_FROM_OBJECT: boolean;
+    static GENERATE_TO_STRING: boolean;
+    static ASSUME_LOCAL_ARRAYS: boolean;
+    /**
+     * Repeated fields numbers.
+     */
+    protected repeatedFields: number[] | undefined | null;
+    /**
+     * Returns the JsPb message_id of this proto.
+     * @return the message id or undefined if this message has no id.
+     */
+    getJsPbMessageId(): string | undefined;
+    /**
+     * Initializes a JsPb Message.
+     * @param msg The JsPb proto to modify.
+     * @param data An initial data array.
+     * @param messageId For response messages, the message id or '' if no message id is specified. For non-response messages, 0.
+     * @param suggestedPivot The field number at which to start putting fields into the extension object. This is only used if data does not contain an extension object already. -1 if no extension object is required for this message type.
+     * @param repeatedFields The message's repeated fields.
+     * @param opt_oneofFields The fields belonging to each of the message's oneof unions.
+     */
+    static initialize(msg: Message, data: any[] | undefined | null, messageId: string | number, suggestedPivot: number, repeatedFields: number[] | null, opt_oneofFields?: number[][] | null): void;
+    /**
+     * Converts a JsPb repeated message field into an object list.
+     * @param field The repeated message field to be converted.
+     * @param toObjectFn The toObject function for this field.  We need to pass this for effective dead code removal.
+     * @param opt_includeInstance Whether to include the JSPB instance for transitional soy proto support: http://goto/soy-param-migration
+     * @return An array of converted message objects.
+     */
+    static toObjectList<T = any>(field: (T | null)[], toObjectFn: ((() => (object | null)) | null) | (() => (object | null)) | null, opt_includeInstance?: boolean): (object | null)[];
+    /**
+     * Adds a proto's extension data to a Soy rendering object.
+     * @param proto The proto whose extensions to convert.
+     * @param obj The Soy object to add converted extension data to.
+     * @param extensions The proto class' registered extensions.
+     * @param getExtensionFn The proto class' getExtension function. Passed for effective dead code removal.
+     * @param opt_includeInstance Whether to include the JSPB instance for transitional soy proto support: http://goto/soy-param-migration
+     */
+    static toObjectExtension(proto: Message, obj: object, extensions: object, getExtensionFn: (this: any) => any, opt_includeInstance?: boolean): void;
```

Comment:
`getExtensionFn: (this: Message, fieldInfo: any) => any`

`fieldInfo` parameter wants to be `ExtensionFieldInfo`, but that would need a type parameter and I've opted for `any` to reduce the complexity there.